nixos/modules/config/xdg/portal.nix

```
@@ -54,13 +54,18 @@ with lib;
     in mkIf cfg.enable {
 
       assertions = [
-        { assertion = (cfg.gtkUsePortal -> cfg.extraPortals != []);
-          message = "Setting xdg.portal.gtkUsePortal to true requires a portal implementation in xdg.portal.extraPortals such as xdg-desktop-portal-gtk or xdg-desktop-portal-kde.";
+        { assertion = cfg.extraPortals != [];
+          message = "Setting xdg.portal.enable to true requires a portal implementation in xdg.portal.extraPortals such as xdg-desktop-portal-gtk or xdg-desktop-portal-kde.";
         }
       ];
 
       services.dbus.packages  = packages;
       systemd.packages = packages;
+      # fixes screen sharing on plasmawayland on non-chromium apps by linking
+      # share/applications/*.desktop files
+      # see https://github.com/NixOS/nixpkgs/issues/145174
+      environment.systemPackages = packages;
```

```suggestion
      environment.systemPackages = [(pkgs.runCommand "xdg-desktop-portal-desktops" {} ''
        mkdir -p "$out/share/applications/"
        cp "${pkgs.xdg-desktop-portal-gtk}/share/applications"/* "$out/share/applications/"
        cp "${pkgs.xdg-desktop-portal-kde}/share/applications"/* "$out/share/applications/"
      '')];
```

And drop the `environment.pathsToLink` bit.

---

`environment.systemPackages` will not link that without `environment.pathsToLink`.